app/src/main/java/com/github/palFinderTeam/palfinder/utils/image/QRCode.kt

```diff
@@ -0,0 +1,68 @@
+package com.github.palFinderTeam.palfinder.utils.image
+
+import android.app.Activity
+import android.content.ContentValues.TAG
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.util.Log
+import androidx.core.content.FileProvider
+import com.ceylonlabs.imageviewpopup.ImagePopup
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+
+
+object QRCode {
+    /**
+     * Saves the image as PNG to the app's private external storage folder.
+     * @param image Bitmap to save.
+     * @return Uri of the saved file or null
+     */
+    private fun saveImageExternal(image: Bitmap, parent: Activity): Uri? {

```

Comment:
Some test could be added for this class

Reply:
yes I will try to add it in my next PR !